bob/src/cluster/quorum.rs

```diff
@@ -30,27 +30,34 @@ impl Quorum {
         debug!("PUT[{}] ~~~PUT LOCAL NODE FIRST~~~", key);
         let mut local_put_ok = 0_usize;
         let mut remote_ok_count = 0_usize;
-        let mut at_least = self.quorum;
+        let at_least = self.quorum;
         let mut failed_nodes = Vec::new();
         let (vdisk_id, disk_path) = self.mapper.get_operation(key);
-        if let Some(path) = disk_path {
+        let local_put_task = disk_path.map(|path| {
             debug!("disk path is present, try put local");
-            let res = put_local_node(&self.backend, key, data, vdisk_id, path).await;
-            if let Err(e) = res {
+            put_local_node(&self.backend, key, data, vdisk_id, path).boxed()
+        });
+        debug!(
+            "PUT[{}] need at least {} additional puts",
+            key,
+            at_least - 1
```

Comment:
This value depends on whether local put started or not. 
I suggest to use `if let Some(path) = disk_path` as it was before and update `at_least` value inside